apps/kol-amelamdim/pages/admin/dashboard/add-weekly-article/index.tsx

```diff
@@ -39,11 +41,75 @@ export const StyledGrid = styled(Grid)`
   }
 `;
 
+const getWeeklyArticleById = async (id: string) => {
+  try {
+    const article = await axios.get(`/api/admin/get-article-by-id?id=${id}`);
+    if (article.data) {
+      return article.data;
+    }
+  } catch (e) {
+    return null;
+  }
+};
+
 const AddWeeklyArticle = () => {
   const [title, setTitle] = useState('');
   const [description, setDescription] = useState('');
   const [weeklyArticleContent, setWeeklyArticleContent] = useState('');
+  const [error, setError] = useState('');
   const router = useRouter();
+  const id = router.query.id;
+
+  const addWeeklyArticle = async () => {
+    try {
+      await axios.post('/api/admin/add-weekly-article', {
+        title,
+        description,
+        content: weeklyArticleContent,
+      });
+
+      setError('');
+      await router.push('/admin/dashboard/list-of-weekly-articles');
+    } catch (e) {
+      setError(API_ERRORS.addWeeklyArticleError.message.heb);
+    }
+  };
+
+  const updateWeeklyArticle = async (id: string) => {
+    try {
+      await axios.post('/api/admin/update-weekly-article', {
+        id,
+        title,
+        description,
+        content: weeklyArticleContent,
+      });
+
+      setError('');
+      await router.push('/admin/dashboard/list-of-weekly-articles');
+    } catch (e) {
+      setError(API_ERRORS.addWeeklyArticleError.message.heb);
+    }
+  };
+
+  const submitHandler = async () => {
+    if (id) {
+      await updateWeeklyArticle(id as string);
+    } else {
+      await addWeeklyArticle();
+    }
+  };
+
+  useEffect(() => {
+    if (id) {
+      getWeeklyArticleById(id as string).then((response) => {
```

Comment:
I think we can do that in getServersideProps function to avoid unnecessary loading

Reply:
right. fixed